src/main/java/pl/put/poznan/analyzer/logic/NetworkAnalyzer.java

```diff
@@ -27,31 +29,57 @@
      */
     private final DFS dfs;
 
+    private final NetworkRepository networkJsonRepository;
+
+    private final NodeListConverter nodeListConverter;
+
     /**
      * Class constructor for spring
+     *
      * @param bfs instance of BFS to be used in program
      * @param dfs instance of DFS to be used in program
      */
     @Autowired
-    public NetworkAnalyzer(BFS bfs, DFS dfs) {
+    public NetworkAnalyzer(BFS bfs, DFS dfs, NetworkRepository networkJsonRepository, NodeListConverter nodeListConverter) {
         this.bfs = bfs;
         this.dfs = dfs;
+        this.networkJsonRepository = networkJsonRepository;
+        this.nodeListConverter = nodeListConverter;
     }
 
     /**
      * Find the most profitable path in the network
+     *
      * @param nodeList network (list of nodes) in which you want to find the best path
-     * @param mode name of algorithm you want to use for searching for the best path
+     * @param mode     name of algorithm you want to use for searching for the best path
      * @return the best path as Result (list of nodes and path's value)
-     *      <br> or NULL if path can't be found
+     * <br> or NULL if path can't be found
      */
     public Result findTheBestPath(List<Node> nodeList, String mode) {
         if (!Data.checkNetwork(nodeList)) {
             logger.error("Incorrect network");
             throw new IllegalArgumentException("Incorrect network");
         }
-        Map <Integer, Node> nodesMap = Data.getNodesMap(nodeList);
+        Map<Integer, Node> nodesMap = Data.getNodesMap(nodeList);
         logger.debug("Prepared to run the algorithm");
         return mode.equals("BFS") ? bfs.run(nodeList) : dfs.run(nodesMap).getResult();
     }
+
+    public int saveNetworkOnDatabase(String nodes) {
+        Network network = new Network(nodes);
+        networkJsonRepository.save(network);
+        return network.getId();
+    }
+
+    public String getNetwork(int id) {
+        Network network = networkJsonRepository.findOne(id);
+        if (network == null) {
+            throw new IllegalArgumentException("Incorrect id");
+        }
+        return network.getJsonValue();
+    }
+
+    public void deleteNetworkFromDatabase(int id) {
+        networkJsonRepository.delete(id);
```

Comment:
Jeśli musi być błąd rzucany przy szukaniu to nie powinno też być informacji przy usuwaniu że szukana baza nie istnieje jeśli jej nie znajdzie?

Reply:
Jest taki błąd, metoda delete zwraca taki błąd

Reply:
Metoda delete z CrudRepository jest typu void, więc niestety nie otrzymamy takiej informacji. Można jeszcze sprawdzić czy baza nie rzuca błędem